test/doubles/test_doubles.go

```diff
@@ -0,0 +1,37 @@
+package testdoubles
+
+import (
+	vcrRecorder "github.com/dnaeon/go-vcr/recorder"
+	commonConfig "github.com/fabric8-services/fabric8-common/configuration"
+	"github.com/fabric8-services/fabric8-tenant/auth"
+	"github.com/fabric8-services/fabric8-tenant/configuration"
+	"github.com/fabric8-services/fabric8-tenant/test/recorder"
+)
+
+func LoadTestConfig() (*configuration.Data, error) {
+	data, err := configuration.GetData()
+	return data, err
+}
+
+func NewAuthClientService(cassetteFile, authURL string, recorderOptions ...recorder.Option) (*auth.Service, *vcrRecorder.Recorder, error) {
+	var options []commonConfig.HTTPClientOption
+	var r *vcrRecorder.Recorder
+	var err error
+	if cassetteFile != "" {
+		r, err = recorder.New(cassetteFile, recorderOptions...)
+		if err != nil {
+			return nil, r, err
+		}
+		options = append(options, commonConfig.WithRoundTripper(r))
+	}
+	config, err := LoadTestConfig()
+	if err != nil {
+		return nil, r, err
+	}
+	config.Set(configuration.VarAuthURL, authURL)
```

Comment:
```
os.Setenv("F8_AUTH_URL", "http://auth.test.com") // check F8_ convention
config.GetAuthURL() // this will return http://auth.test.com
```
so you no need to export `varAuthURL` and also other variable you have exported.

Reply:
I'm just quite afraid of setting the environment variables because I experienced a lot of problems with it in previous projects - mainly when it is possible to do it programmatically.

Reply:
While it's OK to use config.Set() in tests (if you don't some tests in parallel which share the same viper instance. Viper is not thread safe, so, you can't use Set() if the viper instance is used by multiple go routines). But I also don't like exporting var names without a reason (it's not that critical though.. it's more about code cleanness). It's really an implementation detail and configuration consumers don't have to know about that.

When you need to get some custom configuration for you test then you can:

1. config.Set("auth.url", authURL) - it's fine to hardcode the var name in tests. If you change the var name in configuration then this test will fail. So, there is no risk of doing it. And I like this more then exporting the var names.

2. Or you can set the env var as @nurali-techie suggested and reload the config. It's safe if you take care of re-setting it back to the original value. See an example here - https://github.com/fabric8-services/fabric8-auth/blob/master/configuration/configuration_blackbox_test.go#L55-L61
It can be quite verbose though..

Reply:
> Viper is not thread safe

I didn't know that. Anyway, setting system environments is not thread-safe either, is it?

> But I also don't like exporting var names without a reason

ok, make sense - fixed in 3101416 

>  It's safe if you take care of re-setting it back to the original value....
It can be quite verbose though.

yes, the verbosity is the reason why I didn't want to go this way. Currently, it is used in 6 test cases, which means that I would have to copy-paste the same code in all places (technically, I could use test suites or subtests). In addition, as I'm trying to add more this kind of integration tests, where having config initialized is necessary, then the code duplication will be much bigger. But that is only my point of view ;-). I'm open to any solution

Reply:
In this case using config.Set("auth.url", authURL) is the best solution IMO (short and simple). But sometimes you need to test the configuration itself. Then you have to use env vars to test how configuration is loaded.